RInAppMessaging/Classes/CampaignDispatcher.swift

```diff
@@ -39,10 +39,11 @@ internal class CampaignDispatcher: CampaignDispatcherType, TaskSchedulable {
 
         let sessionConfig = URLSessionConfiguration.default
         sessionConfig.timeoutIntervalForRequest = Constants.CampaignMessage.imageResourceRequestTimeoutSeconds
+        sessionConfig.requestCachePolicy = .returnCacheDataElseLoad
```

Comment:
it seems like this setting doesn't help at all.
I ran some tests on this code and it looks like cache works only if `memoryCapacity` or `diskCapacity` are set to 2000% of desired capacity (5% of cached size). So in order for cache to work for 4 images we have to set wither memory or disk size to 400MiB (and yes, we have to count data as a multiple on 1024). As an alternative we could cache images manually using NSCache and use its data instead of making a request if url-key matches

Reply:
Sad we can't use `URLCache` as-is. The problem with `NSCache` is that it's in-memory only. Maybe we can subclass `URLCache` to always store data (up to 100% capacity).

Reply:
I didn't get the same result. From my own testing, it seems like for 5MB files (the apparent max size for the IAM dashboard), if either memoryCapacity or diskCapacity is set to `101*1024*1024` (101MB) then the file will be cached. So 101MB should be the magic number in our use case.

Test:

```swift
import Foundation
import PlaygroundSupport

print("check this dir to see if 2 files were cached:")
print(PlaygroundSupport.playgroundSharedDataDirectory.absoluteString + "/RTest/fsCachedData/")

let conf = URLSessionConfiguration.default
conf.requestCachePolicy = .returnCacheDataElseLoad
//conf.requestCachePolicy = .reloadIgnoringLocalCacheData
conf.urlCache = URLCache(
    memoryCapacity: 512_000 /*default but could be 101MB as well*/,
    diskCapacity: 101*1024*1024,
    diskPath: "RTest")
let session = URLSession(configuration: conf)

let img1 = URL(string: "https://upload.wikimedia.org/wikipedia/commons/2/2d/Snake_River_%285mb%29.jpg")!
let img2 = URL(string: "https://sample-videos.com/img/Sample-jpg-image-5mb.jpg")!

session.dataTask(with: img1) { data,_,err in
    print("done -- ", err?.localizedDescription ?? "success")
}.resume()

session.dataTask(with: img2) { data,_,err in
    print("done -- ", err?.localizedDescription ?? "success")
}.resume()
```